modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java

```diff
@@ -732,6 +732,14 @@ public CodegenOperation fromOperation(String path, String httpMethod, Operation
                 }
                 header.nameInCamelCase = toModelName(header.baseName);
             }
+
+            if (op.authMethods != null) {
+                for (CodegenSecurity s : op.authMethods) {
+                    if (s.isApiKey && s.isKeyInHeader) {
+                        s.vendorExtensions.put("x-keyVarName", toModelName(s.keyParamName));
```

Comment:
If this is really needed, can you use a better name? e.g. `apiKeyName`?

Reply:
I'm open to changing the name if you think that fits better.
As for the needing of this variable is so we can prevent problems like this [example](https://swagger.io/docs/specification/authentication/api-keys/) where the header is `X-API-KEY` which is the value of `keyParamName` but we wouldn't be available to use it like this:

```
// compile error
header! { (X-API-KEY, "X-API-KEY") => [String] }
```

This is the solution I came up with but if you have a better one we can use that one.

Reply:
@richardwhiuk 
Just changed the name from `x-kayVarName` to `x-apiKeyName`